### PR TITLE
🔨 Refactor PHLAT

### PR DESCRIPTION
**Pull request by migbro**
## Description

Some code has been rewritten and rearranged to create a cleaner and functionally equivalent cwl workflow of the desired tool.
The READMEs add context to how it works and decisions made so far.

Fixes # https://github.com/d3b-center/bixu-tracker/issues/2109

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [ ] [Outputs that match local test of original code](https://cavatica.sbgenomics.com/u/d3b-bixu/phlat-hla-dev/files/651c645a07220f0cc33befa7/)
- [ ] [End-to-end Workflow](https://cavatica.sbgenomics.com/u/d3b-bixu/phlat-hla-dev/tasks/4c18a13f-45c3-47c5-91d0-05d8f5c9058b/) creating functionally equivalent output
- [ ] [A second end-to-end test](https://cavatica.sbgenomics.com/u/d3b-bixu/phlat-hla-dev/tasks/06756757-b1ac-4834-a1e5-d653df1efee3/)

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] I have committed any related changes to the PR

### CODE REVIEW

File: docs/D3b_PHLAT_README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# D3b PHLAT Implementation
+
+This is a software revival of sorts based on the the work from [PHLAT: Inference of High-Resolution HLA Types from RNA and Whole Exome Sequencing](https://pubmed.ncbi.nlm.nih.gov/29858810/).
+It was unfortunately implemented in python 2.7 and currently there are no plans to attempt to port over the original code to python3.
+However, the workflow implementation should make this far easier and more accessible to use.
+Some of the support scripts, like `scripts/extensions.py` and `scripts/utilities.py` have been copied over as-is to help ensure functional equivalency, while the former `PHAT.py` (`go()` function replaced by `scripts/phlat_d3b.py`) and `prepare.py` have been replaced with cwltool implementations for ease of use and readability
```
Comment (sickler-alex) [inline docs/D3b_PHLAT_README.md:6]: Make sure `PHLAT` is spelled correctly.

Comment (dmiller15) [inline docs/D3b_PHLAT_README.md:6]: `PHAT.py` sounds cool but is it supposed to be `PHLAT.py`?

File: scripts/phlat_d3b.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+"""
+A rewrite of the PHLAT.py go function to perform final scoring
+"""
```
Comment (dmiller15) [inline scripts/phlat_d3b.py:3]: can you put a shebang here?

**Review by sickler-alex — APPROVED** [review]

File: scripts/utilities.py [hunk 3]
```diff
@@ -1,10 +1,4 @@
-# uncompyle6 version 3.7.4
-# Python bytecode 2.7 (62211)
-# Decompiled from: Python 3.6.8 (default, Apr  2 2020, 13:34:55) 
-# [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
-# Embedded file name: /root/phlat2/dist/utilities.py
-# Compiled at: 2017-09-23 15:54:42
-import pysam, sets, copy, extensions
+import pysam
```
Comment (dmiller15) [inline scripts/utilities.py:1]: shebang pls

**Review by dmiller15 — APPROVED** [review]
Good enough first pass. I'd say the map creation could probably be its own step but that can be addressed if this ever gets more traction for production.

**Pull request merged**